src/include/miopen/tensor.hpp

```diff
@@ -173,6 +175,30 @@ struct TensorDescriptor : miopenTensorDescriptor
 
     std::string ToString() const;
 
+    template <class Vector, class Op>
+    inline std::vector<int64_t> sort_permutation(const Vector& data, Op op) const
+    {
+        std::vector<std::int64_t> result(data.size());
+        std::iota(result.begin(), result.end(), 0);
+        std::sort(
+            result.begin(), result.end(), [&](auto x, auto y) { return op(data[x], data[y]); });
+        return result;
+    }
+
+    std::string GetLayout(std::string labels) const
```

Comment:
This involves creation/deletion of several containers, fillng, sorting etc. Not that fast as a simple getter should be. Shall we rename it to `ComputeLayout()`?

Reply:
We can decide on the name depends on how next review comments goes. Should we keep the function untouched, I shall rename it to `Compute`. Should we decide to cache the result, I will keep as is.

Reply:
I think renaming it to `ComputeLayout` makes sense since it needs to do a caculation.

Reply:
Done, now I have both `ComputeLayout` and `GetLayout`.